### ISSUE TITLE
Release branch with GitHubFlowVersion doesn't append beta tag

### ISSUE THREAD

**Issue by robdmoore** [ParticularLabs/GitVersion#331]
I'm using the latest version of gitversion on chocolatey on my build server and I am building a release branch for a project `release-2.0.0`, but the variables that are generated don't include the beta tag like I would expect from https://github.com/ParticularLabs/GitVersion/wiki/GitHubFlowExamples.

My variables are:

```
##teamcity[buildNumber '2.0.0+49']
[10:58:59]Executing GenerateBuildLogOutput for 'TeamCity'.
[10:58:59]##teamcity[setParameter name='GitVersion.Major' value='2']
[10:58:59]##teamcity[setParameter name='system.GitVersion.Major' value='2']
[10:58:59]##teamcity[setParameter name='GitVersion.Minor' value='0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.Minor' value='0']
[10:58:59]##teamcity[setParameter name='GitVersion.Patch' value='0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.Patch' value='0']
[10:58:59]##teamcity[setParameter name='GitVersion.PreReleaseTag' value='']
[10:58:59]##teamcity[setParameter name='system.GitVersion.PreReleaseTag' value='']
[10:58:59]##teamcity[setParameter name='GitVersion.PreReleaseTagWithDash' value='']
[10:58:59]##teamcity[setParameter name='system.GitVersion.PreReleaseTagWithDash' value='']
[10:58:59]##teamcity[setParameter name='GitVersion.BuildMetaData' value='49']
[10:58:59]##teamcity[setParameter name='system.GitVersion.BuildMetaData' value='49']
[10:58:59]##teamcity[setParameter name='GitVersion.FullBuildMetaData' value='49.Branch.release-2.0.0.Sha.3cc5f402b339e282aa55e739c08a5c36c45f5533']
[10:58:59]##teamcity[setParameter name='system.GitVersion.FullBuildMetaData' value='49.Branch.release-2.0.0.Sha.3cc5f402b339e282aa55e739c08a5c36c45f5533']
[10:58:59]##teamcity[setParameter name='GitVersion.MajorMinorPatch' value='2.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.MajorMinorPatch' value='2.0.0']
[10:58:59]##teamcity[setParameter name='GitVersion.SemVer' value='2.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.SemVer' value='2.0.0']
[10:58:59]##teamcity[setParameter name='GitVersion.LegacySemVer' value='2.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.LegacySemVer' value='2.0.0']
[10:58:59]##teamcity[setParameter name='GitVersion.LegacySemVerPadded' value='2.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.LegacySemVerPadded' value='2.0.0']
[10:58:59]##teamcity[setParameter name='GitVersion.AssemblySemVer' value='2.0.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.AssemblySemVer' value='2.0.0.0']
[10:58:59]##teamcity[setParameter name='GitVersion.AssemblyFileSemVer' value='2.0.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.AssemblyFileSemVer' value='2.0.0.0']
[10:58:59]##teamcity[setParameter name='GitVersion.FullSemVer' value='2.0.0+49']
[10:58:59]##teamcity[setParameter name='system.GitVersion.FullSemVer' value='2.0.0+49']
[10:58:59]##teamcity[setParameter name='GitVersion.InformationalVersion' value='2.0.0+49.Branch.release-2.0.0.Sha.3cc5f402b339e282aa55e739c08a5c36c45f5533']
[10:58:59]##teamcity[setParameter name='system.GitVersion.InformationalVersion' value='2.0.0+49.Branch.release-2.0.0.Sha.3cc5f402b339e282aa55e739c08a5c36c45f5533']
[10:58:59]##teamcity[setParameter name='GitVersion.ClassicVersion' value='2.0.0.49']
[10:58:59]##teamcity[setParameter name='system.GitVersion.ClassicVersion' value='2.0.0.49']
[10:58:59]##teamcity[setParameter name='GitVersion.ClassicVersionWithTag' value='2.0.0.49']
[10:58:59]##teamcity[setParameter name='system.GitVersion.ClassicVersionWithTag' value='2.0.0.49']
[10:58:59]##teamcity[setParameter name='GitVersion.BranchName' value='release-2.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.BranchName' value='release-2.0.0']
[10:58:59]##teamcity[setParameter name='GitVersion.Sha' value='3cc5f402b339e282aa55e739c08a5c36c45f5533']
[10:58:59]##teamcity[setParameter name='system.GitVersion.Sha' value='3cc5f402b339e282aa55e739c08a5c36c45f5533']
[10:58:59]##teamcity[setParameter name='GitVersion.NuGetVersionV2' value='2.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.NuGetVersionV2' value='2.0.0']
[10:58:59]##teamcity[setParameter name='GitVersion.NuGetVersion' value='2.0.0']
[10:58:59]##teamcity[setParameter name='system.GitVersion.NuGetVersion' value='2.0.0']
```

The build is at http://ci.robdmoore.id.au:8010/viewLog.html?buildId=3462&buildTypeId=NTestDataBuilder_CI&tab=buildLog (you can log in as guest).

**Comment by gep13** [comment]
Hmmm, I haven't created a release branch for any of my projects with the new version of GitVersion, but based on the tests, I would have expected this to work:

![image](https://cloud.githubusercontent.com/assets/1271146/5592475/61c597fa-91c9-11e4-9a43-f61820ace32f.png)

i.e. include the ReleaseBranchTag (which defaults to beta) in the Full SemVer Version.

**Comment by JakeGinnivan** [comment]
Figured the problem.. Not sure exactly how to fix it at the moment

NextVersion.txt bumps the version to 2.0.0, the branch name causes the version to be 2.0.0-beta.1

Highest wins.. Deleting NextVersion.txt will give you what you expect though

**Comment by JakeGinnivan** [comment]
Verified this has been fixed in 3.0.0 which will have a beta released shortly

**Comment by robdmoore** [comment]
Nice! Thanks mate :)
